lib/service/supervisor.go

```diff
@@ -145,6 +153,9 @@ func NewSupervisor(id string, parentLog logrus.FieldLogger) Supervisor {
 	closeContext, cancel := context.WithCancel(context.TODO())
```

Comment:
Since we are here, could we tidy this a bit by replacing all the context.TODO() calls with `rootCtx`?

```suggestion
	rootCtx := context.TODO()
	// Use rootCtx where context.TODO() is used.
```